projects/m2/019-morse-code/js/index.js

```diff
@@ -0,0 +1,23 @@
+// translate letters and digits, ignore character not included in the table
+
+const code = { A: '._', B: '_...', C: '_._.', D: '_..', E: '.', F: '.._.', G: '__.', H: '....', I: '..', J: '.___', K: '_._', L: '._..', M: '__', N: '_.', O: '___', P: '.__.', Q: '__._', R: '._.', S: '...', T: '_', U: '.._', V: '.__', W: '.__', X: '_.._', Y: '_.__', Z: '__..', 1: '.____', 2: '..___', 3: '...__', 4: '...._', 5: '.....', 6: '_....', 7: '__...', 8: '___..', 9: '____.', 0: '_____', Á: '.__._', Ä: '._._', É: '.._..', Ö: '___.', Ü: '..__'
+}
+
+const userText= prompt('Enter the message', 'Hello, World!').toUpperCase()
+
+const morseCode = (text, code) => {
+  const splitText = text.split('')
+  let message = ''
+
+   const newMessage = splitText.map((letter) => { 
+    for(let item in code){
+      if(item.includes(letter)){
+        message += code[item] + ' '
```

Comment:
You **complicated the solution**.
Try to evaluate/understand the following possible solution:
<img width="339" alt="image" src="https://user-images.githubusercontent.com/74595044/236615213-dcb2f298-1e8b-4abf-8e18-c7a2a3ee3333.png">